cv-server/src/MockerWebcamREST/mocker_webcam.py

```diff
@@ -0,0 +1,241 @@
+import os
+import signal
+from fastapi import FastAPI, BackgroundTasks
+from pydantic import BaseModel
+from src.Base64.Base64Conversions import *
+import cv2
+import tkinter as tk
+import PIL.Image, PIL.ImageTk
+import threading
+from src.MockerWebcamREST.catbase64 import cat_base64
+from tkinter import filedialog
+from tkinter import font as tkFont
+
+
+class MockerWB(tk.Tk):
+    def __init__(self, *args, **kwargs) -> None:
+        tk.Tk.__init__(self, *args, **kwargs)
+        self.video_capture = VideoCapture()
+        self.gui_setup()
+        self.delay = 10
+        self.playing = False
+        self.webcam = False
+        self.current_frame = None
+        self.ret = False
+        self.server = MockerCaptureCameraServer(self)
+        self.server_thread = threading.Thread(target=self.server.start_server,daemon=True)
+        self.server_thread.start()
+        self.update()
+        
+
+    def gui_setup(self):
+        self.wm_title("Webcam Mocker")
+        self.configure(background="#333333")
+
+        self.top_frame = tk.Frame(self, width=600, height=200, bg='grey')
+        self.top_frame.grid(row=0, column=0, padx=5, pady=5)
+        self.bottom_frame = tk.Frame(self, width=200, height=200, bg='grey')
+        self.bottom_frame.grid(row=1, column=0, padx=5, pady=5)
+
+        helv36 = tkFont.Font(family='Helvetica', size=16, weight=tkFont.BOLD)
+        self.toggle_webcam_btn = tk.Button(
+            self.bottom_frame,
+            text = 'WEBCAM',
+            command = self.toggle_webcam,
+            font = helv36,
+            background = '#7E7E7E'
+        )
+        self.open_file_btn = tk.Button(
+            self.bottom_frame,
+            text = 'OPEN',
+            command = self.select_file,
+            font = helv36
+        )
+        self.play_btn = tk.Button(
+            self.bottom_frame,
+            text = 'PLAY',
+            command = self.play_video,
+            font = helv36
+        )
+        self.stop_btn = tk.Button(
+            self.bottom_frame,
+            text = 'STOP',
+            command = self.stop_video,
+            font = helv36
+        )
+        self.replay_btn = tk.Button(
+            self.bottom_frame,
+            text = 'REPLAY',
+            command = self.replay_video,
+            font = helv36
+        )
+        self.video_canvas = tk.Canvas(
+            self.top_frame,
+        )
+        self.video_canvas.grid(row=0, column=0)
+        self.toggle_webcam_btn.grid(row=0, column=0, padx=5, pady=5, ipadx=10)
+        self.open_file_btn.grid(row=0, column=1, padx=5, pady=5, ipadx=10)
+        self.replay_btn.grid(row=0, column=2, padx=5, pady=5, ipadx=10)
+        self.play_btn.grid(row=0, column=3, padx=5, pady=5, ipadx=10)
+        self.stop_btn.grid(row=0, column=4, padx=5, pady=5, ipadx=10)
+
+
+    def update(self):
+        if self.video_capture.video != None and self.playing:
+            ret, frame = self.video_capture.get_frame()
+            self.ret = ret
```

Comment:
Czy self.ret jest potrzebny? A jak już jest potrzebny to czy nie można po prostu _self.ret, frame = ..._

Reply:
No potrzebuje jak unity woła o nowe klatki to żebym wiedział czy mogę je wyświetlić i przesłać czy mam zapchać używając cat_base64